OpenRA.Game/Graphics/SpriteCache.cs

```diff
@@ -136,13 +116,18 @@ public void LoadReservations(ModData modData)
 			// We can achieve better sheet packing by keeping sprites with similar heights together.
 			var orderedPendingResolve = pendingResolve.OrderBy(x => x.Frame.Size.Height);
 
-			var spriteCache = new Dictionary<(string Filename, int FrameIndex, bool Premultiplied), Sprite>(pendingResolve.Count);
-			foreach (var (filename, frameIndex, premultiplied, frame, spritesForToken) in orderedPendingResolve)
+			var spriteCache = new Dictionary<(
+				string Filename,
+				int FrameIndex,
+				bool Premultiplied,
+				Func<ISpriteFrame, ISpriteFrame> AdjustFrame),
+				Sprite>(pendingResolve.Count);
+			foreach (var (filename, frameIndex, premultiplied, adjustFrame, frame, spritesForToken) in orderedPendingResolve)
 			{
 				// Premultiplied and non-premultiplied sprites must be cached separately
 				// to cover the case where the same image is requested in both versions.
 				spritesForToken[frameIndex] = spriteCache.GetOrAdd(
-					(filename, frameIndex, premultiplied),
+					(filename, frameIndex, premultiplied, adjustFrame),
```

Comment:
adjustFrame is a reference type, so the filtering won't work perfectly when non-null. Though I suppose this implementation is still much cleaner than on bleed